src/components/form/Form.tsx

```diff
@@ -31,6 +31,11 @@ interface FormState {
 export function Form() {
   const currentPageId = useCurrentView();
   const { isValidPageId, navigateToPage } = useNavigatePage();
+  const { searchParams, clearSearchParam } = useNavigationParams();
+  const componentId = searchParams.get(SearchParams.FocusComponentId);
+  const focusNode = useResolvedNode(componentId);
+  const navigateTo = useNavigateToNode();
+  const location = useLocation();
```

Comment:
Did you check if this made `<Form />` re-render more often? I know I struggled a bit to reduce the amount of hooks this component uses in order to minimize re-renders, as re-rendering this might trigger re-rendering of all components (which is costly).

Reply:
It could also be possible to do this in a component like the `<ErrorProcessing />` component (declared below here), which doesn't render any children below it, so it won't affect performance of `<Form />` when only it has to re-render.

Reply:
Yep, creating a new component for this was what I meant - I see now that my suggestion didn't make that entirely clear.. 🙈